src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3244,6 +3261,8 @@ internal override StrongNameKeys StrongNameKeys
             bool hasDeclarationErrors = !FilterAndAppendDiagnostics(diagnostics, GetDiagnostics(CompilationStage.Declare, true, cancellationToken), excludeDiagnostics, cancellationToken);
             excludeDiagnostics?.Free();
 
+            hasDeclarationErrors |= CheckDuplicateInterceptions(diagnostics);
```

Comment:
From offline discussion, maybe this should be moved into `CompileMethodBodies` and maybe `CheckDuplicateFilePaths` as well.
Then the `ERR_ModuleEmitFailure` reported in `TryGetInterceptor` becomes unnecessary.

Reply:
Also consider adding tests for duplicate interceptions and duplicate file paths in a compilation for ref-only assembly (showing whether or not we ran those checks).

Reply:
This comment also seems relevant.

https://github.com/dotnet/roslyn/pull/67786#discussion_r1171811141

Basically, the search is for a consistent place to report duplicates at the end of the declaration phase. This place would be appropriate for both duplicate file-local types and for interceptors.